### PR TITLE
Add Metering support to the admin settings

### LINKED ISSUE #3564: Integrate metering into KKP dashboard
The information provided by the [metering](https://github.com/kubermatic/metering) tool should be accessible from the dashboard.

The metering tool collects data about the usage of user clusters (CPU, RAM, ...). This data shall be provided in the form of CSV file downloads. A list of reports (daily, weekly and/or monthly reports) should be shown to the user, who can choose to download them individually.

### LINKED ISSUE #3589: Create a UI scheme for metering
**User Story** 
As a user I would like to enable the metering tool in all seed cluster in the UI, so I don't have to enable it using the command line. 

**Acceptance Criteria**
- The admin panel has a new side bar option which would take the user to configure metering tool.
- The metering tool is disabled by default.
- There should be two types of input fields, the credentials and the configurations.
- The credentials are: bucket, accessKey, secretKey, endpoint all of type string 
- The configurations are: enabled(boolean), storageClassName(string), storageClassName(string).

For the API Calls, I will update the ticket once another PR will be merged.

### PR DESCRIPTION
**Pull request by floreks**
### What this PR does / why we need it
metering

### Which issue(s) this PR fixes

Closes #3589 
Closes #3564 

### Release note

```release-note
Add metering support to the EE version of KKP Dashboard
```

### CODE REVIEW

File: src/app/core/services/metering.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  constructor(private readonly _http: HttpClient) {}
+
+  saveConfiguration(configuration: MeteringConfiguration): Observable<any> {
+    const url = `${this._restRoot}/admin/metering/configurations`;
+    return this._http.put(url, configuration);
+  }
+
+  saveCredentials(credentials: MeteringCredentials): Observable<any> {
+    const url = `${this._restRoot}admin/metering/credentials`;
```
Comment (kgroschoff) [inline src/app/core/services/metering.ts:33]: / is missing ;)

File: src/app/core/module.ts [hunk 2]
```diff
@@ -28,6 +28,7 @@ import {EndOfLifeService} from '@core/services/eol';
 import {GlobalModule} from '@core/services/global/module';
 import {HistoryService} from '@core/services/history';
 import {LabelService} from '@core/services/label';
+import {MeteringService} from '@core/services/metering';
```
Comment (kgroschoff) [inline src/app/core/module.ts:31]: I'm wondering if we also want to keep the `MeteringService` inside the `dynamic/enterprise` folder, as it is an ee-only feature?
On the other hand for sure it is looking cleaner to have all services in one place. 🤔 

Just a thought 🤔

File: src/app/dynamic/enterprise/metering/config/credentials-dialog/template.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  <mat-spinner *ngIf="saving"
+               [diameter]="25"
+               @pushToSide
+               color="accent"
+               class="spinner"></mat-spinner>
+
+  <button mat-flat-button
+          (click)="save()"
+          [disabled]="form.invalid || saving">
+    Edit Credentials
```
Comment (kgroschoff) [inline src/app/dynamic/enterprise/metering/config/credentials-dialog/template.html:66]: Please add `<i class="km-icon-mask km-icon-edit"></i>` to the edit button, to align with Chiaras button improvements.

File: src/app/dynamic/enterprise/metering/config/config-dialog/template.html [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  <mat-spinner *ngIf="saving"
+               [diameter]="25"
+               @pushToSide
+               color="accent"
+               class="spinner"></mat-spinner>
+
+  <button mat-flat-button
+          (click)="save()"
+          [disabled]="form.invalid || saving">
+    Configure Metering
```
Comment (kgroschoff) [inline src/app/dynamic/enterprise/metering/config/config-dialog/template.html:49]: Please also add `<i class="km-icon-mask km-icon-edit"></i>`

**Comment by kgroschoff** [comment]
I hope it is okay, that I've already took a look 😉  
Overall it is looking good so far 😃

File: src/app/core/module.ts (hunk 2 above)
Comment (floreks) [reply src/app/core/module.ts:31]: Ye, this is a good idea. I'll move it.

**Review by kgroschoff — COMMENTED** [review]
/lgtm

**Pull request merged**